### ISSUE TITLE
Images

### ISSUE THREAD

**Issue by YouriT** [phillipj/node-plex-api#4]
Hi,

First of all, nice work, really appreciated thanks for it :D

I was just wondering how do you retrieve & display images ? When requesting through the API it seams to work but the received data doesn't seem to be directly displayed as a JPEG, the binary seems to be broken somewhere.

Example code:
```js
client.query('/library/metadata/6782/art/1426945195').then(function(result) {
	res.writeHead(200, {
		'Content-Type': 'image/jpeg'
	});
	res.end(result);
}, function(err) {
	throw new Error('Could not connect to server');
});
```

The result seems to be a binary, I even see a JFIF somewhere in the header.

Thanks !

**Comment by YouriT** [comment]
Well it was just a bit tricky because of the request automatic encoding in utf8 which was making everything not working.

I got it working by adding after line 100 in api.js:
```
    if (relativeUrl.indexOf('art') > -1 ||
        relativeUrl.indexOf('thumb') > -1 ||
        relativeUrl.indexOf('banner') > -1) {
        reqOpts.encoding = null;
    }
```

Solves the issue and now I can do:
```
client.query('/library/metadata/6782/art/1426945195').then(function(result) {
	res.writeHead(200, {
		'Content-Type': 'image/jpeg'
	});
	res.end(result);
}, function(err) {
	throw new Error('Could not connect to server');
});
```

Without any problem :)

**Comment by phillipj** [comment]
Hi @YouriT, thanks for reporting and debugging this!

To be honest I havent retrieved images before, so didnt think of the default utf8 encoding as an issue. Seeing your example makes me think of streaming right away. It could have looked something like:

```js
client.query('/library/metadata/6782/art/1426945195', { stream: true })
    .pipe(res)
    .on('error', function onError(){
        res.writeHead(500);
        res.end('Couldnt connect to Plex server :(');
    });
```

Any thoughts about that?

**Comment by YouriT** [comment]
Well that could be an idea. Or we could rather detect the response header to see if the content-type is image or sound or whatever and if it's JSON, encode it and respond it. As it no options needed no ?

**Comment by phillipj** [comment]
Sure, that would atleast fix the biggest issue here

**Comment by YouriT** [comment]
Will check that this evening :)

**Comment by YouriT** [comment]
I did a pull request to implement this feature (#5) :)
It's not perfect because it considers any other responses than application/json to be a stream but I don't see any other possible case so if needed we can make the difference but I don't find that very useful for now no ?

**Comment by YouriT** [comment]
If you accept it, could you update the npm registry please ? Thx a lot !

**Comment by phillipj** [comment]
Fixed with your commit cc1f60f889fcc
